Added section on high-volume change processing (#80)
neo4j/docs-cdc: modules/ROOT/pages/examples/index.adoc

Note that `db.cdc.current` needs to be called and stored before calling `db.cdc.query` for this to work.

Another way to avoid this issue is to use broader selectors and perform client side filtering instead.

[[monitoring-and-address-cdc-client-performance]]

> == Monitoring and addressing CDC client performance

Reviewer: ```suggestion
== High-volume change processing
```
This is more descriptive

Author: I think we should stick with "Monitoring and addressing CDC client performance" since it reflects the changes made to address your previous comment.

> For optimal performance when the change rate is high, first monitor whether your CDC client is keeping up with Neo4j.

Reviewer: ```suggestion
For optimal performance when the change rate is high, it is important to monitor your CDC client's performance to ensure that the client is keeping up with Neo4j.
```
It says "...*first* monitor whether..." but this is not followed by a second step? So I assume that you monitor the lag and if it is deemed a problem, remedies are described?

Author: First monitor, then address performance. And you should definitely monitor first... there's no need to address an issue which doesn't exist.

Track lag by comparing the latest processed `txId` with the latest `txId` in Neo4j, and/or by monitoring `txCommitTime`.
If lag keeps increasing, apply the recommendations below.

> For high change rates, a single dedicated reader thread is usually sufficient for a CDC stream.

Reviewer: Is the point here that a single *dedicated* reader thread is sufficient? Or is the point that a *single* thread is enough (as in multiple reader threads are NOT needed)?

Author: What's the difference? I don't want to make a stronger statement in the abstract.

The critical requirement is that this reader does not pause for downstream processing.

Use a hand-off design:

* One dedicated reader continuously consumes the CDC stream and immediately hands off each change to an in-memory queue.
* A pool of worker threads consumes from the queue and performs processing in parallel.

Avoid short polling patterns such as "fetch X events, stop, then reconnect".
`db.cdc.query` streams continuously until the client catches up, and keeping a single stream open is typically the fastest approach.
Repeatedly stopping and restarting the stream adds overhead and reduces throughput.

The Neo4j driver buffers incoming changes in memory.
If the reader does not hand off fast enough, that buffer can fill up and the server will stop sending additional changes until space is available.
Similarly, if the application queue fills up, worker backpressure will block the reader and propagate the same slowdown.

> Plan capacity across the full pipeline:

Reviewer: ```suggestion
If the reader does not hand off fast enough, that buffer can fill up and the server will stop sending additional changes until space is available.
Similarly, if the application queue fills up, worker backpressure will block the reader and propagate the same slowdown.
Therefore, plan capacity across the full pipeline:
```
